### PR TITLE
front: update selected projection after train deletion

### LINKED ISSUE #5403: front: projection is not updated when a train is deleted from the list
A timetable with two trains is created. The two trains don't share any part of their path. If the train who determines the projection chosen is deleted, the GET is empty.

Normally the projection should be updated with the path of the other train. 

Version 358b34a7

### PR DESCRIPTION
**Pull request by clarani**
closes #5403

### CODE REVIEW

**Review by Akctarus — APPROVED** [review]
LGTM

File: front/src/utils/array.ts [hunk 1]
```diff
@@ -1,3 +1,4 @@
-// eslint-disable-next-line import/prefer-default-export
 export const isFirstOrLastElement = <T>(array: T[], element: T) =>
   array[0] === element || array[array.length - 1] === element;
+
+export const distinctElements = <T>(array: T[]) => [...new Set(array)];
```
Comment (SharglutDev) [inline front/src/utils/array.ts]:
```suggestion
export const uniqueElements = <T>(array: T[]) => [...new Set(array)];
```

File: front/src/applications/operationalStudies/components/Scenario/getSimulationResults.ts [hunk 2]
```diff
@@ -33,8 +34,10 @@ export default async function getSimulationResults(
       store.dispatch(updateSelectedTrainId(trainSchedulesIDs[0]));
     }
 
+    const timetablePaths = distinctElements(timetable.train_schedule_summaries.map((trainSchedule) => trainSchedule.path_id));
+
```
Comment (SharglutDev) [inline front/src/applications/operationalStudies/components/Scenario/getSimulationResults.ts]: There is a warning here

**Review by SharglutDev — CHANGES_REQUESTED** [review]
Thank you for this PR ! Left two comments. 

Tested and it works if the remaining train has a different path, it gets the projection.

But if we have 3 trains, 2 with same path and one different, if we delete one of the "common path" train, the projection isn't set on any train in the list.

File: front/src/utils/array.ts (hunk 1 above)
Comment (clarani) [reply front/src/utils/array.ts]: Renamed :heavy_check_mark:

File: front/src/applications/operationalStudies/components/Scenario/getSimulationResults.ts (hunk 2 above)
Comment (clarani) [reply front/src/applications/operationalStudies/components/Scenario/getSimulationResults.ts]: Thanks !

**Review by SharglutDev — APPROVED** [review]
Lgtm, tested in local, very nice job :)

**Review by nicolaswurtz — APPROVED** [review]
👍

**Review by Akctarus — APPROVED** [review]
Tested in local, LGTM

**Pull request merged**